pages/add-rides/index.js

```diff
@@ -0,0 +1,51 @@
+import React, { useState } from "react";
+import RideAddForm from "../../components/Form/RideAddForm.styled";
+import Heading from "../../components/Heading/Heading.styled";
+import SecondHeading from "../../components/SecondHeading/SecondHeading.styled";
+import { StyledPaymentInformation } from "../../components/Card/Card.styled";
+
+const formatDate = (dateString) => {
+  const date = new Date(dateString);
+  const options = {
+    day: "2-digit",
+    month: "2-digit",
+    year: "numeric",
+  };
+  const formattedDate = date.toLocaleDateString("de-DE", options);
+  const [dateStringFormatted] = formattedDate.split(", ");
+  return `${dateStringFormatted}`;
+};
+
+const RideAddPage = ({ onAddRide, rides }) => {
+  const [isRideAdded, setIsRideAdded] = useState(false);
+  const handleAddRide = (ride) => {
+    const newRide = {
+      ...ride,
+      slug: `ride-${rides.length + 1}`,
+      currency: "€",
+      meetingpoint: "",
+      date: formatDate(ride.date),
+      time: ride.time + "Uhr",
+      isMyRide: true,
+    };
```

Comment:
You are constructing a new object here with a `slug` but your component that maps the rides seems to expect an `id` (as a key). Both are fine but you'll need to have a unique id.
Counting the length of the array is not a great solution since it will break once you delete rides inbetween creating rides. In this case, ids will be assigned twice.

I'd suggest to create a unique id.
```bash
npm i uid
```

At the top of the file import it.
```js
import { uid } from 'uid';
``` 

```suggestion
    const id = uid();
    const newRide = {
      ...ride,
      id,
      slug: `ride-${id}`,
      currency: "€",
      meetingpoint: "",
      date: formatDate(ride.date),
      time: ride.time + "Uhr",
      isMyRide: true,
    };
```